### DISCUSSION TITLE
How to validate something in the OnChange method?

### DISCUSSION THREAD

**Discussion by HadiSDev** [react-hook-form/react-hook-form#2536] [Q&A] [answered]
Hi,

I am trying to make an input only accept integers as input:

```
            <Input
              value={value}
              name="test"
              type="number"
              ref={formMethods.register}
              onChange={(event: any) => {
                console.log(event.target.value);
                if (isInt(event.target.value)) {
                  setValue(event.target.value);
                } else {
                  return;
                }
              }}
            />
```

However, because onChange is overridden from the register, this won't work. Is there a workaround for this problem?

**Comment by bluebill1049** [comment, marked as answer]
https://react-hook-form.com/api/#setValue shouldValidate: true

**Reply by mvvmm** [reply]
@bluebill1049 Is it possible to only validate in accordance with the `mode`? 

For example, if I have:

```jsx
const {...} = useForm({
    mode: "onTouched",
    ...
});
```

Can I do something like:
```jsx
setValue("myValue", "5", { shouldValidate: 'sameAsMode' });
```

or even: 
```jsx
setValue("myValue", "5", { shouldValidate: 'onTouched' });
```

This would validate only if that field has `isTouched = true`, same as the `mode` dictates.

I can of course add this logic myself:
```jsx
setValue(
    "myValue",
    "5",
    {
        shouldValidate: getFieldState("myValue").isTouched
             ? true
             : false,
    },
);
```

I'm just curious if there is a more elegant solution.

**Comment by HadiSDev** [comment]
@bluebill1049 Is it not possible to override the OnChange method, without using controlled input?

**Reply by bluebill1049** [reply]
no, you can use `onChange`, but not with `value`, that will turn it into controlled input.
